frontend/model/contracts/shared/time.js

```diff
@@ -7,6 +7,48 @@ export const HOURS_MILLIS = 60 * MINS_MILLIS
 export const DAYS_MILLIS = 24 * HOURS_MILLIS
 export const MONTHS_MILLIS = 30 * DAYS_MILLIS
 
+const plusOnePeriodLength = (timestamp: string, periodLength: number): string => (
+  dateToPeriodStamp(addTimeToDate(timestamp, periodLength))
+)
+
+const minusOnePeriodLength = (timestamp: string, periodLength: number): string => (
+  dateToPeriodStamp(addTimeToDate(timestamp, -periodLength))
+)
+
+// @parameter knownSortedStamps:
+// - MUST be sorted in ascending order (lower timestamps first)
+// - MAY NOT contain duplicate elements
+export function periodStampsForDate (
+  date: Date | string,
+  { knownSortedStamps, periodLength }: { knownSortedStamps: string[], periodLength: number }
+): Object {
+  if (typeof date === 'string' && !isIsoString(date)) {
```

Comment:
You're right, and I would even remove the `date instanceof Date` check if possible, since Flow annotations should already handle that.
Initially I added the `isIsoString` check only because string was are a superset of ISO strings

Edit: After further testing, it appears that `.toISOString()` throws when called on invalid date objects like new Date('foo'). So to avoid possible uncaught errors here we'll have to do a bit more validation or use a `try` block...

Or maybe we could restrict the API to just not accept Date objects at all?

Reply:
However Flow complains about `if (!(isIsoString(date) || date instanceof Date)`, and changing the check order didn't do anything to it

Reply:
In this suggestion I used `&&` rather than` ||` to avoid an extra pair or parens, and `isNaN `to guard against `Invalid Date` values. These are quite uncommon but could cause the `toISOString()` call to throw

Reply:
> used && rather than || to avoid an extra pair or parens

It's better to have the parens than to have an extra logical check imo

> and isNaN to guard against Invalid Date values.

I really don't think that's needed

> Otherwise it's equivalent to if (!(isIsoString(date) || date instanceof Date))

It's not... the `||` condition results in 1 check most of the time, whereas the modified `&&` results in 3 different check every time